modules/deployment-scripts/create-kv-certificate/main.bicep

```diff
@@ -0,0 +1,143 @@
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbacKv 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = {
+  name: guid(akv.id, rbacRolesNeededOnKV, useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id)
+  scope: akv
+  properties: {
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', rbacRolesNeededOnKV)
+    principalId: useExistingManagedIdentity ? existingDepScriptId.properties.principalId : newDepScriptId.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource createImportCert 'Microsoft.Resources/deploymentScripts@2020-10-01' = [for certName in certNames: {
+  name: 'AKV-Cert-${akv.name}-${replace(replace(certName,':',''),'/','-')}'
+  location: location
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${useExistingManagedIdentity ? existingDepScriptId.id : newDepScriptId.id}': {}
+    }
+  }
+  kind: 'AzureCLI'
+  dependsOn: [
+    rbacKv
+  ]
+  properties: {
+    forceUpdateTag: forceUpdateTag
+    azCliVersion: azCliVersion
+    timeout: timeout
+    retentionInterval: retention
+    environmentVariables: [
+      {
+        name: 'RG'
+        value: resourceGroup().name
+      }
+      {
+        name: 'akvName'
+        value: akvName
+      }
+      {
+        name: 'certName'
+        value: certName
+      }
+      {
+        name: 'initialDelay'
+        value: initialScriptDelay
+      }
+      {
+        name: 'spClientId'
+        value: useExistingManagedIdentity ? existingDepScriptId.properties.clientId : newDepScriptId.properties.clientId
+      }
+    ]
+    scriptContent: '''
+      #!/bin/bash
+      set -e
+
+      echo "Waiting on Identity RBAC replication ($initialDelay)"
+      sleep $initialDelay
+
+      #echo "Querying role assignments for service principal"
+      #az role assignment list --assignee $spClientId
+
+      echo "Creating AKV Cert $certName"
+      az keyvault certificate create --vault-name $akvName -n $certName -p "$(az keyvault certificate get-default-policy | sed -e s/CN=CLIGetDefaultPolicy/CN=${certName}/g )";
+
+      echo "Getting akv secretid for $certName";
+      versionedSecretId=$(az keyvault certificate show -n $certName --vault-name $akvName --query "sid" -o tsv);
+      unversionedSecretId=$(echo $versionedSecretId | cut -d'/' -f-5) # remove the version from the url;
+
+      jsonOutputString=$(jq -n --arg cn "$certName" --arg sid "$unversionedSecretId" --arg vsid "$versionedSecretId" '{certName: $cn, certSecretId: {versioned: $vsid, unversioned: $sid }}')
+      echo $jsonOutputString > $AZ_SCRIPTS_OUTPUT_PATH
+    '''
+    cleanupPreference: cleanupPreference
+  }
+}]
+
+@description('Array of info from each Certificate Deployment Script')
+output createdCertificates array = [for (certName, i) in certNames: {
```

Comment:
another thought -- we should output the thumbprint too... many scenarios that need certs need that and there's no way to retrieve it via the control plane.  It does mean, the script would have to poll to wait until the cert is created.

Another sample: https://github.com/bmoore-msft/AzureRM-Samples/blob/master/ignite-2019/Create-Cert.ps1#L17

Reply:
added the thumbprint as output.

also wrapped this in a loop, but the `az keyvault certificate create` operation seems to be synchronous - so the loop seems unnecessary.